### DISCUSSION TITLE
The problem with fullscreen

### DISCUSSION THREAD

**Discussion by zethon** [zethon/ttvg#161] [General]
On my MacBook Pro, it looks fine when I use it as a laptop, but once I use my Thunderbolt this is what I see:

![image](https://user-images.githubusercontent.com/5203583/165058845-999c25a6-bd78-4ee2-b288-ee70cda006bc.png)

How do other games work on Ubuntu? We need to figure out how to fix this across the board, but I'm not sure what the fix is.

**Comment by zethon** [comment]
Actually, my bad, even with the old code I still see the same thing. Yeah, I'm not sure how to fix this.

**Reply by charmyt** [reply]
This is similar to what I saw on Ubuntu, before adding this fix:

```
sf::VideoMode::getDesktopMode()
```

Except on Ubuntu the "little" fullscreen was in the top left corner, as opposed to centered.

I'm wondering if on some platforms, we shouldn't be creating a new instance of sf::VideoMode, but should instead use one of the existing instances of that class. (Even though the newly created instance has the same height/width/bitdepth as one of the already available instances.)

There is a mechanism for listing all available fullscreen modes.

https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1VideoMode.php

```
std::vector<sf::VideoMode> modes = sf::VideoMode::getFullscreenModes();
```

Not sure if that helps. 

Maybe this is similar to the sfml quirkiness about putting texture instances into vectors. Like maybe these VideoModes that are already made available, under the hood contain pointers to lower level resources, which new instances created at the application layer will not have?
